dockstore-integration-testing/src/test/java/io/dockstore/client/cli/OrganizationIT.java

```diff
@@ -1683,6 +1685,93 @@ public void testDuplicateAliasOperations() {
         }
     }
 
+    /**
+     * Tests that we are getting the number of workflows correctly
+     */
+    @Test
+    public void testWorkflowsLength() {
+        // Setup user who creates Organization and collection
+        final ApiClient webClient = getWebClient(USER_2_USERNAME, testingPostgres);
+        OrganizationsApi organizationsApi = new OrganizationsApi(webClient);
+
+        //set up admin user
+        final ApiClient webClientAdminUser = getWebClient(ADMIN_USERNAME, testingPostgres);
+        OrganizationsApi organizationsApiAdmin = new OrganizationsApi(webClientAdminUser);
+
+        //manually register and then publish this workflow
+        WorkflowsApi workflowApi = new WorkflowsApi(webClient);
+        Workflow workflow = workflowApi
+                .manualRegister("github", "dockstore-testing/viral-pipelines", "/pipes/WDL/workflows/multi_sample_assemble_kraken.wdl", "", "wdl",
+                        "");
+        final Workflow workflowByPathGithub = workflowApi.getWorkflowByPath("github.com/dockstore-testing/viral-pipelines", null, false);
+
+        workflowApi.refresh(workflowByPathGithub.getId(), false);
+        workflowApi.publish(workflowByPathGithub.getId(), CommonTestUtilities.createPublishRequest(true));
+
+        // Create the Organization and collection
+        Organization organization = createOrg(organizationsApi);
+        Collection stubCollection = stubCollectionObject();
+
+        long orgId = organization.getId();
+
+        // Attach collections
+        Collection collection = organizationsApi.createCollection(orgId, stubCollection);
+
+        long collectionId = collection.getId();
+
+        // Approve the org
+        organizationsApiAdmin.approveOrganization(organization.getId());
+
+        // Add tool to collection
+        organizationsApi.addEntryToCollection(orgId, collectionId, workflow.getId(), null);
+
+        Collection addedCollection = organizationsApi.getCollectionById(orgId, collectionId);
+        long workflowsCount = addedCollection.getWorkflowsLength();
+        assertEquals(1, workflowsCount);
+    }
```

Comment:
Can you also test the pinned versions we were discussing?

Reply:
Have been trying for a while, commit 2a1e7730809be73f0558ad2b3c58b0ed5c1e68c4 has my work in progress, seems like everything is working just fine but only fails at assertion, if anyone can help

Reply:
Realized most tools do not have a second version, and because I had the exact same code for getting number of tools and number of workflows, I didn't test the versions in `testToolsLength`, but if it's needed I will add that in